compiler/pom.xml

```diff
@@ -173,6 +156,41 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <version>2.3</version>
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <minimizeJar>true</minimizeJar>
+              <artifactSet>
+                <excludes>
+                  <!-- guava which has a consistent API and whose public types we vend in producers -->
+                  <exclude>com.google.guava</exclude>
```

Comment:
FYI, Immutables.org shades Guava but uses [a trick](https://github.com/immutables/immutables/blob/5360e8677440ab13e797f94ea6b2160734ea6d8a/value-processor/src/org/immutables/value/processor/meta/UnshadeGuava.java) ([example use](https://github.com/immutables/immutables/blob/5360e8677440ab13e797f94ea6b2160734ea6d8a/value-processor/src/org/immutables/value/processor/meta/AttributeTypeKind.java)) to avoid the maven-shade-plugin from rewriting Guava class name literals used in generated code.